### DISCUSSION TITLE
XYChart Legend

### DISCUSSION THREAD

**Discussion by JamesIrish** [airbnb/visx#1143] [Q&A] [answered]
Hi team,

Firstly thanks for the awesome work on visx!  Question: how do I add a legend to an XYChart? 

For instance on the gallery of the XYChart https://airbnb.io/visx/xychart how would I add an ordinal legend to the bottom (underneath the X axis) that had the three series listed horizontally (San Fran, NY, Austin) and a coloured dash to present the line color?

I've looked into the legends package https://airbnb.io/visx/legends but I don't understand how to access the scale instance/objects that are required for the legend and are buried within the XYChart somewhere!

Any pointers appreciated, 
Thanks,
James

**Comment by williaster** [comment, marked as answer]
Hey @JamesIrish 👋  thanks for checking out `visx`. Here's a demo incorporating a [legend](https://codesandbox.io/s/xychart-legend-demo-mgqr7?file=/Example.tsx) into `XYChart`. The key is the `DataContext` which is described under `AdvancedUsage` in the [`@visx/xychart` docs](https://airbnb.io/visx/docs/xychart).

Legends were initially on the MVP features for `@visx/xychart` but didn't make the cut. Thankfully it's pretty straight forward to integrate. I left additional comments in the sandbox. Note this is an HTML legend (which is what `@visx/legend` is based on), so you can't render it as a child of `XYChart` since the chart is SVG based. Hope that helps!

![image](https://user-images.githubusercontent.com/4496521/113917249-d6869e80-9795-11eb-99d8-b5838720cbf9.png)

**Reply by JamesIrish** [reply]
Perfect!!  Thanks Chris, great example.
